driver-core/src/main/java/com/datastax/driver/core/CodecRegistry.java

```diff
@@ -219,7 +201,65 @@ public int hashCode() {
         }
     }
 
-    private final boolean immutable;
+    /**
+     * A complexity-based weigher.
+     * Weights are computed according to the CQL type:
+     * <ol>
+     * <li>Primtive types weigh 0;
+     * <li>Collections weigh the total weight of their inner types + the weight of their level of deepness;
+     * <li>UDTs and tuples weigh the total weight of their inner types + the weight of their level of deepness, but cannot weigh less than 1;
+     * <li>Custom (non-CQL) types weigh 1.
+     * </ol>
+     * A consequence of this algorithm is that primitive types and all "shallow" collections thereof
+     * are never evicted.
+     */
+    private static class TypeCodecWeigher implements Weigher<CacheKey, TypeCodec<?>> {
```

Comment:
From Guava's docs, it seems that they wouldn't be size evicted, but they _could_ be evicted for other reasons (capacity, access time etc.). So I think it's reasonable not to worry about it, besides users also have the possibility to provide their own LoadingCache instance if they have particular needs.